plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java

```diff
@@ -145,6 +156,15 @@ static String buildImage(@Nonnull DockerClient dockerClient,
       buildParameters.add(DockerClient.BuildParam.noCache());
     }
 
+    if (buildArgs != null && !buildArgs.isEmpty()) {
+      try {
+        final String encodedBuildArgs = URLEncoder.encode(new Gson().toJson(buildArgs), "utf-8");
```

Comment:
I was surprised by this as well but without the encoding it complains about `"` in the request:
```
[INFO] [ERROR] Failed to execute goal com.spotify:dockerfile-maven-plugin:1.3.2-ooyala-build-args:build (default) on project basic-with-build-args: Execution default of goal com.spotify:dockerfile-maven-plugin:1.3.2-ooyala-build-args:build failed: Illegal character """ at position 21 is not allowed as a start of a name in a path template "pull=true&buildargs={"IMAGE_VERSION":"0.0.1"}". -> [Help 1]
```